### PR TITLE
2018-01-09 Hwancheol : Back-end service frame

### PR DESCRIPTION
**Pull request by khc196**

### CODE REVIEW

File: back-end/app.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+## !/usr/bin/env python
+"""
+Author : Hwancheol Kang
+Project : Alpha-search
+Data : 5th, January, 2018
```
Comment (shberry) [inline back-end/app.py]: Date?

Comment (shberry) [reply back-end/app.py]: Also double check on the date format. Looks a bit strange :)

File: back-end/app.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from __future__ import print_function
+
+import json
+import os
+
+from flask import Flask
+from flask import request
+from flask import make_response
+
+import uuid
```
Comment (shberry) [inline back-end/app.py]: is it possible to group all imports together?

File: back-end/app.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+# Generate Redis Object
+redis_obj = redis.Redis(host='localhost',port=6379,db=0)
+
+# This route is for fulfillment webhook for an Dialogflow agent.
+@app.route('/webhook', methods=['POST'])
+def webhook():
+    # Get json from dialogflow. 
+    req = request.get_json(silent=True, force=True)
+
+    print("Request:")
```
Comment (shberry) [inline back-end/app.py]: if print statements are temporary, add a TODO to remove these later :)

File: back-end/app.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    print(json.dumps(req, indent=4))
+
+    res = processWebhookRequest(req)
+
+    # JSON Encoding
+    res = json.dumps(res, indent=4)
```
Comment (shberry) [inline back-end/app.py]: maybe we can create a constant (something like indent = 4) that holds the value 4 and use that constant for indents? In this case, it's easy to update when indent space count changes somehow :)

File: back-end/app.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    r = make_response(res)
+    r.headers['Content-Type'] = 'application/json'
+    return r
+
```
Comment (shberry) [inline back-end/app.py:46]: remove new line :)

**Review by shberry — COMMENTED** [review]
Cool :) Thanks!

**Comment by khc196** [comment]
@shberry Thank you for your review. I've updated!

**Review by dexterastin — APPROVED** [review]
Cool!

**Pull request merged**